kerl

```diff
@@ -754,7 +668,133 @@ _flags() {
+    os_release_id0=$(\grep "^ID=" "$1")
+    os_release_id=$(echo "${os_release_id0}" | \sed 's|.*=||')
+    os_release_pretty_name0=$(\grep "^PRETTY_NAME=" "$1")
+    os_release_pretty_name=$(echo "${os_release_pretty_name0}" | \sed 's|.*=||')
+    echo "[packages] Found ${os_release_pretty_name} with ID ${os_release_id}" >>"$LOGFILE"
+    if [ -z "${os_release_id}" ]; then
+        echo "[packages] Release ID not found in $1" >>"$LOGFILE"
+        echo "2"
+    else
+        echo "[packages] Release ID found in $1: ${os_release_id}" >>"$LOGFILE"
+        echo "${os_release_id}"
+    fi
+}
+
+get_id_from_os_release_files() {
+    # returns:
+    # 1 - if no release files exist
+    files="/etc/os-release /usr/lib/os-release"
+    for file in ${files}; do
+        if [ -f "${file}" ]; then
+            parsed=$(parse_os_release "${file}")
+            if [ "${parsed}" != "2" ]; then
+                break
+            fi
+        fi
+    done
+
+    if [ -z "${parsed}" ]; then
+        echo "[packages] Found no file in: ${files}. Bailing out..." >>"$LOGFILE"
+        echo "1"
+    fi
+
+    echo "${parsed}"
+}
+
+probe_pkgs() {
+    os_release_id=$(get_id_from_os_release_files)
+
+    if [ "${os_release_id}" = "1" ]; then
+        msg="[packages] Unable to determine Linux flavor (no release files); not checking build packages."
+        echo "${msg}" >>"$LOGFILE"
+        l=w stderr "${msg}"
+        return
+    elif [ "${os_release_id}" = "2" ]; then
+        msg="[packages] Unable to determine Linux flavor (no ID); not checking build packages."
+        echo "${msg}" >>"$LOGFILE"
+        l=w stderr "${msg}"
+        return
+    fi
+
+    kpp=$(eval echo \$_KPP_"${os_release_id}"_pkgs)
+    if [ -n "${kpp}" ]; then
+        echo "[packages] Found package declarations for ${os_release_id}. Linux flavor is known." >>"$LOGFILE"
+        echo "[packages] Acting on known Linux distro ${os_release_id}" >>"$LOGFILE"
+        for pkg in ${kpp}; do
+            cmd=$(eval echo "\$_KPP_${os_release_id}_probe")
+            probe_without_1=$(echo "${cmd}" | sed "s|\$1||")
+            if [ "${probe_without_1}" != "${cmd}" ]; then
+                cmd=$(echo "${cmd}" | sed "s|\$1|${pkg}|")
+            fi
+            eval "${cmd}" >/dev/null 2>&1
+            probe_res=$?
+            if [ "${probe_res}" != 0 ]; then
+                msg="[packages] Probe failed for ${pkg} (distro: ${os_release_id}): \"${cmd}\" returned ${probe_res}"
+                echo "${msg}" >>"$LOGFILE"
+                l=w stderr "${msg}"
+            else
+                echo "[packages] Probe success for ${pkg} (distro: ${os_release_id})!" >>"$LOGFILE"
+            fi
+        done
+    else
+        msg="[packages] Unknown Linux flavor ${os_release_id}; not checking build packages."
+        echo "${msg}" >>"$LOGFILE"
+        l=w stderr "${msg}"
+    fi
+}
+
 _do_build() {
+    LOGFILE="$KERL_BUILD_DIR/$2/otp_build_$1.log"
+
```

Comment:
Because `probe_pkgs` needs it 😄

Reply:
@jadeallenx, I wonder if it'd make sense to reset to logfile to empty, at this point.

It was previously (and it still compatible with)

```
    LOGFILE="$KERL_BUILD_DIR/$2/otp_build_$1.log"
    # Add a build date and env in logfile
    echo "*** $(date) - kerl build $1 ***" >>"$LOGFILE"
```

but this means we have several build logs in this file, forever growing. Maybe it's intended (?)